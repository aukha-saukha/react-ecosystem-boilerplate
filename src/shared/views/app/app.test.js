import { /* mount, */ shallow } from 'enzyme';
import React from 'react';
// import { MemoryRouter } from 'react-router-dom';

import App from './app.react';
// import NotFound from '../not-found';

function mockUserStore() {
  const original = require.requireActual('../../../client/utilities/user-store');

  return {
    ...original,
    getSettingFromUserStore: jest.fn(),
  };
}

jest.mock('../../../client/utilities/user-store', () => mockUserStore());

describe('App component', () => {
  // Disabling this test for now because in Enzyme, useEffect is not called in shallow render.
  // Bug https://github.com/airbnb/enzyme/issues/2086 is open.
  // TODO: Fix it once Enzyme supports it.
  // eslint-disable-next-line jest/no-commented-out-tests
  // it('Invalid path renders "Not Found" page', () => {
  //   const wrapper = mount(
  //     <MemoryRouter initialEntries={['/random']}>
  //       <App />
  //     </MemoryRouter>
  //   );
  //   expect(wrapper.find(NotFound)).toHaveLength(1);

  //   wrapper.unmount();
  // });

  it('Renders correctly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
