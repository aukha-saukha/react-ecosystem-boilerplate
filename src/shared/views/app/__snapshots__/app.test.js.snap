// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders correctly 1`] = `
<ErrorBoundary>
  <div
    className="wrapper"
  >
    <Header />
    <section
      className="padding-all-15 page-specific-content"
    >
      <Switch>
        <Route
          exact={true}
          key="/about-us"
          path="/about-us"
          render={[Function]}
        />
        <Route
          exact={true}
          key="/contact-us"
          path="/contact-us"
          render={[Function]}
        />
        <Route
          exact={true}
          key="/"
          path="/"
          render={[Function]}
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </section>
    <Footer />
  </div>
</ErrorBoundary>
`;
